src/services/infra/InfraService.ts

```diff
@@ -127,6 +132,19 @@ export default class InfraService {
     return url
   }
 
+  isValidUrl(url: string): boolean {
+    const schema = Joi.string().domain()
+    if (
+      schema.validate(url).error &&
+      // joi reports initial "_" for certificates as as an invalid url WRONGLY,
+      // therefore check if after removing it, it reports as a valid url
+      schema.validate(url.substring(1)).error
+    ) {
+      return false
+    }
+    return true
```

Comment:
could we define a variable like so: `const extractedUrl = url.startsWith('_') ? url.substring(1) : url` and then do `return schema.validate(extractedUrl).error`

Reply:
sure 611e2d1